eosclubhouse/clubhouse.py

```diff
@@ -91,68 +92,70 @@
 resource = Gio.resource_load(os.path.join(config.DATA_DIR, 'eos-clubhouse.gresource'))
 Gio.Resource._register(resource)
 
-CharacterInfo = {
-    'estelle': {
-        'position': (610, 144),
-        'username': 'lightspeedgal',
-        'pathway': 'art',
-        'pathway_title': 'Art'
-    },
-    'ada': {
-        'position': (204, 182),
-        'username': 'countesslovelace',
-        'pathway': 'games',
-        'pathway_title': 'Games'
-    },
-    'saniel': {
-        'position': (892, 570),
-        'username': 'srowe1822',
-        'pathway': 'operatingsystem',
-        'pathway_title': 'Operating Systems'
-    },
-    'faber': {
-        'position': (518, 511),
-        'username': 'fabersapiens',
-        'pathway': 'maker',
-        'pathway_title': 'Maker'
-    },
-    'riley': {
-        'position': (298, 551),
-        'username': '_getriled',
-        'pathway': 'web',
-        'pathway_title': 'Web'
-    },
-    'felix': {
-        'position': None,
-        'username': 'UNDEFINED_USER',
-        'pathway': None,
-        'pathway_title': None
-    },
-    'endless': {
-        'position': None,
-        'username': 'Endless',
-        'pathway': None,
-        'pathway_title': None
-    },
-    'daemon': {
-        'position': None,
-        'username': 'cron',
-        'pathway': None,
-        'pathway_title': None
-    },
-    'daemon2': {
-        'position': None,
-        'username': 'peacock',
-        'pathway': None,
-        'pathway_title': None
-    },
-    'daemon3': {
-        'position': None,
-        'username': 'starman',
-        'pathway': None,
-        'pathway_title': None
+
+def CharacterInfo():
```

Comment:
Can you explain why this has to be a function now?

Reply:
This is needed because the gettext is initialized in the `eos-clubhouse` script but after importing the config, because we need the locales path, so if we've this as a variable in the global module level, this is initialized before the locales, so the text is not translated correctly.

With this change the content is not initialized at import, is created on every call so the translated strings are created after the locales is set correctly.

Reply:
OK, thanks for the explanation. I never liked this CharacterInfo that got introduced in Hack2. It's data so it should't be in the python code, it would be better to read from a csv or json file.

Anyways, wouldn't be better to use gettext when the info is read by the Character class? Something like:

```
            self.pathway_title = _(info['pathway_title'])
```

Reply:
The gettext strings are extracted from the code to generate the .po file, so it should be literal strings. In other case the extractor needs to run the actual code to get the final string to translate and I'm not sure that that should work.

Reply:
Ah, right. The proper way to do this would be: a) move data to a JSON, b) extract translatable text from JSON directly, similar to what happens in the UI files. There are tools like pybabel for that. But let's keep the hacky way for now.